aws_doc_sdk_examples_tools/labels/from_doc_gen.py

```diff
@@ -0,0 +1,135 @@
+        expanded=expanded,
+        version=str(doc_gen_service.version),
+        labels=labels,
+    )
+    return service
+
+
+def _examples(examples: Dict[str, DocGenExample]) -> Set[Example]:
+    return set([_example(id, example) for id, example in examples.items()])
+
+
+def _example(id: str, doc_gen_example: DocGenExample) -> Example:
+    labels = doc_gen_example_labels(doc_gen_example)
+
+    example = Example(
+        id=id,
+        title=doc_gen_example.title,
+        title_abbrev=doc_gen_example.title_abbrev,
+        synopsis=doc_gen_example.synopsis or "",
+        synopsis_list=doc_gen_example.synopsis_list,
+        labels=labels,
+    )
+
+    return example
+
+
+def doc_gen_example_labels(doc_gen_example):
+    labels: List[Label] = []
+    if doc_gen_example.category:
+        labels.append(Label(name=known_labels.CATEGORY, value=doc_gen_example.category))
+    if doc_gen_example.service_main:
+        labels.append(Label(name=known_labels.SERVICE, value=doc_gen_example.service_main))
+    for service, actions in doc_gen_example.services.items():
+        labels.append(Label(name=known_labels.SERVICE, value=service))
+        for action in actions:
+            labels.append(Label(name=known_labels.ACTION, value=action))
+    return labels
+
+
+def _snippets(doc_gen_examples: Iterable[DocGenExample], doc_gen_snippets: Dict[str, DocGenSnippet]) -> Set[Snippet]:
+    snippets = set()
+    for example in doc_gen_examples:
+        example_labels = doc_gen_example_labels(example)
+        for lang, language in example.languages.items():
+            lang = lang.lower()
+            for version in language.versions:
+                excerpts: List[Excerpt] = []
+                if version.block_content:
+                    excerpts.append(Excerpt(description=version.block_content))
+                for excerpt in version.excerpts:
+                    if excerpt.description:
+                        excerpts.append(Excerpt(description=excerpt.description))
+                    for tag in [*excerpt.snippet_tags, *excerpt.snippet_files]:
+                        doc_gen_snippet = doc_gen_snippets.get(tag, None)
+                        if doc_gen_snippet:
+                            excerpts.append(Excerpt(path=doc_gen_snippet.file, range=(doc_gen_snippet.line_start, doc_gen_snippet.line_end), content=doc_gen_snippet.code))
+                labels: List[Label] = [
+                    *example_labels,
+                    Label(name=known_labels.SDK, value=f"{lang}:{version.sdk_version}"),
+                ]
+                for service in version.add_services:
+                    labels.append(Label(name=known_labels.SERVICE, value=service))
+                snippet = Snippet(
+                    id=f"{example.id}:{lang}:{version.sdk_version}",
+                    labels=labels,
+                    excerpts=excerpts
+                )
+                snippets.add(snippet)
+
+    return snippets
```

Comment:
I think the new "snippet" is more or less a recast of "example", but now "example" doesn't carry any content to go with it's metadata.

Reply:
Agreed that the naming isn't right. As we discussed offline, we'll do a rename in a follow up PR.